platforms/core-runtime/base-services/src/main/java/org/gradle/internal/reflect/JavaMethod.java

```diff
@@ -116,11 +116,11 @@ public boolean isStatic() {
         return Modifier.isStatic(method.getModifiers());
     }
 
-    public R invokeStatic(Object... args) {
+    public R invokeStatic(@Nullable Object... args) {
```

Comment:
❓ Does it really make sense to allow the varargs array to be `null`?

Reply:
See the to-be-merged nullability guide: https://github.com/gradle/bt-internal-docs/pull/195/files#diff-8b3efc1ff952518669eac5f6c4bc666f6cec3660200be372d521635c8e72ac39R97-R102

Reply:
Thanks for the reference, @octylFractal.

I don't think I understand the benefits of adding the annotation to vararg array parameters after reading it though. Isn't it still desirable to communicate that the array reference itself cannot be null?

Reply:
Maybe I can add another clarification here.

When the `@Nullable` annotation is `TYPE_USE` (as is the case with JSpecify), then the following syntax `@Nullable Object... args` and also `@Nullable Object[] args` means "the array reference cannot be null, but array elements can be null`.

The `javax` annotation is not `TYPE_USE`, and because of this we cannot communicate exactly what we want with it. So the next best thing is to keep the annotation in the same place, signalling the intention of nullability with respect to the method implementations.